### DISCUSSION TITLE
CKFinder integration.

### DISCUSSION THREAD

**Discussion by smileBeda** [Studio-42/elFinder#3478] [Q&A]
I do not understand if elFinder allows CK Editor users to actually circumvent the _purchase_ of a CKFinder license.

It seems that CKEditor makes it 100% impossible to actually pick from already uploaded files when inserting files into the editor.
This is a big drawback and I am looking for a _free_ open source solution that allows to _browse_ already uploaded files to _insert those into the editor_.
Just like simple upload of CKEditor does, but that one only allows you to _upload_ new files. Of course no user is going to re-upload each image over and over again. They will want to _choose from already uploaded files_.

Does elFinder integration make this possible?

**Comment by nao-pon** [comment]
I'm not familiar with CKFinder, so I'm not sure if elFinder will replace it.

However, you can integrate elFinder with CKEditor to insert existing images and files and upload images and files.

Live demo
- [With CKEditor4](https://hypweb.net/elFinder-nightly/demo/2.1/ckeditor.htm)
- [With CKEditor5](https://hypweb.net/elFinder-nightly/demo/2.1/ckeditor5.html)

**Comment by smileBeda** [comment]
Now I’m confused 😆
The very wiki of elFinder says to build on top of CKFinder: https://github.com/Studio-42/elFinder/wiki/Integration-with-CKEditor-5

anyway, I’m supper happy to not use ckfinder
Do you have a guide or so on how you integrated that in that demo?
Otherwise I’ll just look at the page source, should be ok too to get me started.

thanks!

**Comment by smileBeda** [comment]
The sample for CK5 shared above, is clearly using `ckfinder`.

That, is a paid tool of creditor. It's not free - thus, you cannot use it for free.
It is officially a paid plugin: https://ckeditor.com/docs/ckeditor5/latest/features/images/image-upload/ckfinder.html:
> CKFinder is a commercial solution. This feature is enabled by default in all [predefined builds](https://ckeditor.com/docs/ckeditor5/latest/installation/advanced/alternative-setups/predefined-builds.html) for convenience, but the editor will still work properly without activating it.

Also from the build:
<img width="764" alt="Screenshot 2022-04-26 at 21 05 45" src="https://user-images.githubusercontent.com/11800236/165318272-f0abb572-d5a0-4305-9269-e28e4ce5b045.png">

Is this somehow another plugin with the precise same name?
(I can clearly see it working locally too, after using your source example, but... they say it will stop at some point on their doc, unless you subscribe.)

**Comment by nao-pon** [comment]
The online demo sample does not use CKFinder, but uses [ckeditor5-adapter-ckfinder](https://github.com/ckeditor/ckeditor5/tree/master/packages/ckeditor5-adapter-ckfinder) included in CKEditor and is integrated with elFinder.

**Comment by smileBeda** [comment]
This is not what I see.
When you build CKFinder with https://ckeditor.com/ckeditor-5/online-builder/, and _exclude_ CKFinder, you will get JS errors saying CKFinder was not found, because the integration JS Code from the demo explicitly requires that command: `ckfinder`:
`const ckf = editor.commands.get('ckfinder'),`

This Wiki additionally states the same: https://github.com/Studio-42/elFinder/wiki/Integration-with-CKEditor-5

It works in your test site because you use the default build, in which case ckFinder is always included, it however is not free.

**Reply by nao-pon** [reply]
I made a mistake in the pointer. The correct name is [ckeditor5-ckfinder](https://github.com/ckeditor/ckeditor5/tree/master/packages/ckeditor5-ckfinder).

In the demo sample, it uses the ckfinder plugin only to display the toolbar icon, and clicking it does not create instance of CKFinder and does not run CKFinder.

I couldn't find a clear document as to whether this usage would be a licensing issue.

My wishful thinking is that CKFinder is paid, but the ckfinder plugin that launches it is not. Therefore, if you do not enable CKFinder, you may have no problem using the ckfinder plugin. Well, this is my selfish speculation.

Please ask the CKFinder side about using the ckfinder plugin in such a way that you never enable CKFinder. And if you have any problems with such usage, please let us know.

**Reply by smileBeda** [reply]
I’ve reached out to them and am expecting reply soon.
I’ll then update here with findings.

How old is that demo? If it’s older than 30 days it means it’s still working and thus should be ok.

thanks!

**Reply by nao-pon** [reply]
The wiki page was created in 17 Jan 2019, so it's been working with similar code since then.  It doesn't access the CKFinder cloud, so I don't think there's anything wrong with the expiration date.
